src/run_build.py

```diff
@@ -63,11 +74,11 @@ def main() -> int:
             architecture=args.architecture or manifest.build.architecture,
         )
 
-        build_recorder = BuildRecorder(target)
+        build_recorder = BuildRecorder(target, build_manifest) if args.incremental else BuildRecorder(target)
 
         logging.info(f"Building {manifest.build.name} ({target.architecture}) into {target.output_dir}")
 
-        for component in manifest.components.select(focus=args.components, platform=target.platform):
+        for component in manifest.components.select(focus=components, platform=target.platform):
```

Comment:
If `component` is empty then `select()` will return all components here. Need to take care of it here.

Reply:
Need to exit if there is no component changed.

Reply:
Resolved in the latest commit. Please take a look thanks.